src/sql/base/browser/ui/table/hybridDataProvider.ts

```diff
@@ -0,0 +1,135 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AsyncDataProvider, IObservableCollection } from 'sql/base/browser/ui/table/asyncDataView';
+import { FilterableColumn } from 'sql/base/browser/ui/table/interfaces';
+import { CellValueGetter, TableDataView, TableFilterFunc, TableSortFunc } from 'sql/base/browser/ui/table/tableDataView';
+import { IDisposableDataProvider } from 'sql/base/common/dataProvider';
+import { Event, Emitter } from 'vs/base/common/event';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+
+export interface HybridDataProviderOptions {
+	inMemoryDataProcessing: boolean;
+	inMemoryDataCountThreshold?: number;
+}
+
+export class HybridDataProvider<T extends Slick.SlickData> implements IDisposableDataProvider<T> {
```

Comment:
More comments here would be good. What is this a hybrid of? How does that tie in to the in-memory stuff?